File: Tarea 1/Mínimo común múltiplo/mcm.pl
use Scalar::Util qw(looks_like_number);
use Data::Types qw(is_int);
use List::Util qw(max);

sub input{
	print "Ingresar numeros deseados uno por uno (0 para terminar): ";
	while($cantidad != 0){
		$cantidad = <>;
		if(looks_like_number($cantidad)){
			if(is_int($cantidad)){
				if($cantidad == 0){
					last;
				}
				elsif($cantidad > 0){
					if(repetido($cantidad) == 1){
						print "No repetir numeros\n";
					}
					elsif($cantidad == 1){
						print "Numero omitido\n";
					}
					else{
						push(@numeros, $cantidad);
					}
					
					redo;
				}
			}
		}
		print "Solo numeros enteros positivos\n";
		$cantidad = 1;
	}
}

sub repetido{
	my ($num) = @_;
	foreach my $val (@numeros){
		if($val == $num){
			return 1;
		}
	}
	
	return 0;
}

$cantidad = 1;
print "Calculadora del minimo comun multiplo de 'n' numeros\n";
$len = 3;
do{
	print "Se requieren al menos dos numeros\n\n";
	input();
	$len = @numeros;
}
while($len < 2);

print "mcm(";
foreach $num (@numeros){
	chomp $num;
	print "$num, ";
}
print "\b\b  \b\b) = ";

$mcm = max @numeros;
$mcm--;
do{
	$mcm++;
	$esMcm = 1;
	foreach $num (@numeros){
		if($mcm % $num != 0){
			$esMcm = 0;
			last;
		}
	}
}
while($esMcm == 0);

print "$mcm";

File: Tarea 1/Mínimo común múltiplo/mcm.py
def entrada():              #leer números y guardarlos en el arreglo
    n = 1                       #el número a guardar
    print("Ingresar los números deseados uno por uno (0 para terminar)")
    while n != 0:               #mientras no se ingrese 0
        n = input()             #leer número
        if n == "":            #si no se introdujo nada
            continue                #volver a leer
        if not n.isdigit():   #si no es un número
            print("Solo números enteros positivos")
            continue
        else:                   #si sí es número
            n = int(n)              #convertir entrada a entero
            if n == 0:              #si se ingresa 0
                continue                #repetir ciclo
            elif n == 1:
                print("Número omitido")
                continue

            if not repetido(n):    #si no está repetido
                nums.append(n)          #agregar número al arreglo
            else:                   #si está repetido
                print("No repetir números")
    return

def repetido(nuevo):    #checar si el número ya se había introducido
    for num in nums:        #recorrer cada número del arreglo
        if num == nuevo:        #si hay coincidencia
            return True                #regresar 1 (sí está repetido)
    return False                #regresar 0 (no está repetido)

nums = []               #arreglo para guardar números
print("Calculadora del mínimo común múltiplo")
largo = 0               #tamaño del arreglo, de momento es 0 
while largo < 2:        #mientras el tamaño sea menor de 2
    print("Se requieren al menos dos números")
    entrada()               #leer números y guardarlos en arreglo
    largo = len(nums)       #obtener largo del arreglo

#imprimir 'mcm(a, b, c...) ='
print("mcm(", end="")
for x in nums:
    print(x,", ", end="", sep="")
print("\b\b  \b\b", end="")
print(") = ", end="")

mcm = max(nums)     #obtener el número mayor del arreglo
esMcm = 0           #de momento no es mcm
while esMcm == 0:   #mientras no sea mcm
    esMcm = 1           #de momento sí es mcm
    for x in nums:      #recorrer cada número del arreglo
        if(mcm % x != 0):   #si el residuo entre el supuesto mcm y el número es diferente de 0
            esMcm = 0           #no es mcm
            break
    if(esMcm == 1):     #si sí es mcm
        break
    mcm += 1            #incremento de mcm

print(mcm)
